qa/common/check_copyright.py

```diff
@@ -105,34 +109,42 @@ def visit(path):
                 + path + ": " + line)
             return False
 
-        start_year = 0
-        end_year = 0
+        years = []
 
         m = single_re.match(line[len(prefix):])
         if m and len(m.groups()) == 1:
-            start_year = end_year = int(m.group(1))
+            years.append(int(m.group(1)))
         else:
             m = range_re.match(line[len(prefix):])
             if m and len(m.groups()) == 2:
-                start_year = int(m.group(1))
-                end_year = int(m.group(2))
+                years.append(int(m.group(1)))
+                years.append(int(m.group(2)))
             else:
-                print("copyright year is not recognized for " + path + ": " +
-                      line)
-                return False
-
-        if start_year > FLAGS.year:
-            print("copyright start year greater than current year for " +
-                  path + ": " + line)
-            return False
-        if end_year > FLAGS.year:
-            print("copyright end year greater than current year for " + path +
-                  ": " + line)
-            return False
-        if end_year < start_year:
-            print("copyright start year greater than end year for " + path +
+                copyright_row = line[len(prefix):]
+                if new_range_re.match(copyright_row):
+                    year_str = copyright_row.split("(c) ")[1].split(
+                        " NVIDIA ")[0]
+                    for year in year_str.split(","):
+                        if len(year) == 4:  # 2021
+                            years.append(int(year))
+                        elif len(year) == 2:  # 21
+                            years.append(int(year) + 2000)
+                        else:  # 21-23
+                            years.append(int(year[0:2]) + 2000)
+                            years.append(int(year[3:5]) + 2000)
+                else:
+                    print("copyright year is not recognized for " + path +
+                          ": " + line)
+                    return False
+        if years[0] > FLAGS.year:
```

Comment:
This checks that the start year is greater than the current year, and that the the years are increasing, but does it check whether the end year is greater than the current year?

Reply:
Updated